theblues/charmstore.py

```diff
@@ -375,3 +379,16 @@ def fetch_macaroon(self):
             charmstore_url=self.url)
         response = self._get(url)
         return response.text
+
+
+def _get_path(entity_id):
+    '''Get the entity_id as a string if it is a Reference.
+
+    @param entity_id The ID either a reference or a string of the entity
+          to get.
+    @returns: entity_id as a string
+    '''
+    if isinstance(entity_id, references.Reference):
```

Comment:
Not a review, but just glanced at this, please use isinstance only when really necessary:
```python
try:
    path = entity_id.path()
except AttributeError:
    path = entity_id
return path
```